opendevin/core/logger.py

```diff
@@ -10,6 +10,7 @@
 
 DISABLE_COLOR_PRINTING = False
 DEBUG = os.getenv('DEBUG', 'False').lower() in ['true', '1', 'yes']
+LOG_TO_FILE = os.getenv('LOG_TO_FILE', 'False').lower() in ['true', '1', 'yes']
```

Comment:
Can you please tell why is this necessary, is the actual issue that when this file is imported, it runs code? That is, if it didn't run its initialization at import, would it work fine?

Reply:
Yes, so whenever `opendevin` is imported, it will try to by default creates `logs/` directory under the project root, which is kinda not necessary for client running inside runtime, and for server running inside the `opendevin app` docker. 

Whenever the project root has a different write permission for the current user, the `mkdir logs/` will fail and break the code. Since imo `logs/` will primarily be accessible and useful during debugging session, we can just set it as an optional behavior, rather than default behavior, which allows us to simplify other part a bit - wdyt?

Reply:
OK, I'm [refactoring out](https://github.com/OpenDevin/OpenDevin/pull/3413/files#diff-c6567843b80b57fcc1d61101cf9fd68a5fcf75f3f24ae4236d1dce64b4531f5fR286) the initialization during import, and also plan to make the log directory for those llm logs configurable. Maybe it will also address this? To try to replicate it, I should run the `app` docker with a different user then, that's the only way it happens?

I don't know about removing logs by default. We sometimes ask for logs from users to figure out what happened, and people are also asking to see LLM prompts. But maybe I'm missing something, I didn't try it yet!

Reply:
I'm also thinking to split the runtime log from the general `opendevin.log`, so what comes from docker basically, to be in its own place. Currently the general file has everything except llm prompts, it's a lot, specially now with the new runtime. What do you think?

Reply:
That could complicate things a bit to keep an eye on both, so to speak. The docker logs can be a bit spammy, but especially in case of issues, they're worth it and pretty "convenient" to have them in the same log _at the right chronological order_.

Reply:
Maybe we could reduce the amount of Docker log lines, though, by only including the newest lines since the last log grab (and not always "the last 10")? 🤔

Reply:
This is only the files, they are still in console. Although... wait, no, not split the runtime log, but make it possible to enable/disable or set to debug separately from the rest.

Reply:
Yes, that sounds good

Reply:
yeah, i think overall have a good way to control when/how we can log is the ideal way! I'd still hope the logging is by default false (e.g., when people import opendevin, they won't expect opendevin will create a new folder in their disk). We can add env var to enable this in the docker, if we really want this behavior -- but the issue is, even if that's enabled, the user will have hard-time accessing those logs, since they are stored *inside* the docker container.